libethash-cl/CLMiner.cpp

```diff
@@ -858,6 +854,26 @@ bool CLMiner::initEpoch_internal()
                 m_dag.push_back(
                     cl::Buffer(m_context[0], CL_MEM_READ_ONLY, (m_epochContext.dagSize) / 2));
             }
+            cllog << "Creating light cache buffer, size: "
+                  << dev::getFormattedMemory((double)m_epochContext.lightSize);
+            m_light.clear();
+            bool light_on_host = false;
+            try
+            {
+                m_light.push_back(
+                    cl::Buffer(m_context[0], CL_MEM_READ_ONLY, m_epochContext.lightSize));
+            }
+            catch (cl::Error const& err)
+            {
+                // Ok, no room for light cache on GPU. Try allocating on host
+                clog(WarnChannel) << "No room on GPU, allocating light cache on host";
+                clog(WarnChannel) << "Generating DAG will take minutes instead of seconds";
+                light_on_host = true;
+                m_light.clear();
+            }
+            if (light_on_host)
+                m_light.push_back(cl::Buffer(m_context[0], CL_MEM_READ_ONLY | CL_MEM_ALLOC_HOST_PTR,
```

Comment:
This also may throw with CL_OUT_OF_RESOURCES or CL_OUT_OF_HOST_MEMORY. Maybe enclose in a try .. catch block and handle the impossibility to allocate light.

Reply:
This is already enclosed in a catch/try block.